### PR TITLE
Morden: Fix Gallery Alignment Issue causing from setting max-width in desktop

### PR DESCRIPTION
**Pull request by itowhid06**
Fixes https://github.com/Automattic/themes/issues/2144.

#### Changes proposed in this Pull Request:
This PR fixes the image gallery alignment issue in Morden theme after applying https://github.com/Automattic/themes/pull/2077.

**Gallery with left alignment Before Fix**

image

**Gallery with Left alignment After Fix**
image

This PR also fixes the similar issue with right aligned gallery.

#### Related issue(s):
https://github.com/Automattic/themes/issues/2065

### CODE REVIEW

**Comment by itowhid06** [comment]
Hello @allancole,

We can use `max-width: $content-width-xxl` but in that case the left or right aligned galleries are not completely aligned to left or right. I have added the screenshots for above solutions.

**Left Aligned Gallery**

image

**Right Aligned Gallery**

image

Please let me know if that's an acceptable solution.

**Comment by allancole** [comment]
Yeah that doesn’t look correct. I think I have an idea…

File: morden/sass/_extra-child-theme.scss [hunk 1]
```diff
@@ -25,7 +25,6 @@ a {
 
 	@include media(desktop) {
 		box-shadow: 0 0 30px rgba(#000, 0.2);
```
Comment (allancole) [inline morden/sass/_extra-child-theme.scss:27]: Let’s bring back the `$content-width-xxl` variable and recompile the styles.

```suggestion
		box-shadow: 0 0 30px rgba(#000, 0.2);
		max-width: $content-width-xxl;
```

Additionally, if you add the following under the Blocks section of `morden/sass/_extra-child-theme.scss` (around line 250), it should fix the margins for gallery and all blocks that need this same floating behavior.

```
/* 
 * Blocks
 */

// Constrain left-aligned blocks
@media only screen and (min-width: 1280px) {
	.entry-content > .wp-block-button:not(.alignleft):not(.alignright),
	.entry-content > .alignleft {
		margin-left: calc( 0.5 * (100% - calc( 782px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}

// Constrain right-aligned blocks
@media only screen and (min-width: 1280px) {
	.entry-content > .alignright {
		margin-right: calc( 0.5 * (100% - calc( 782px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}
```

**Comment by itowhid06** [comment]
Hello @allancole,

I have included the suggested changes. Please have a look.

File: morden/sass/_extra-child-theme.scss [hunk 2]
```diff
@@ -247,6 +247,24 @@ body {
...
+}
+
+// Constrain right-aligned blocks
+@media only screen and (min-width: 1280px) {
+	.entry-content > .alignright {
+		margin-right: calc( 0.5 * (100% - calc( 782px - 32px)));
+		max-width: calc( 0.5 * calc( 782px - 32px));
+	}
+}
+
```
Comment (allancole) [inline morden/sass/_extra-child-theme.scss:267]: I forgot to see how this affected the other break points. Sorry about that. 

The following should fix the issue for all screen sizes. It’s a pretty big chunk of code but it fixes the issue properly. Make that change and I’ll go ahead and get this merged. 

```suggestion

// Add space for floated content
@media only screen and (min-width: 560px) {
	.entry-content > .wp-block-button:not(.alignleft):not(.alignright),
	.entry-content > .alignleft {
		margin-right: #{map-deep-get($config-global, "spacing", "horizontal")};
	}

	.entry-content > .alignright {
		margin-left: #{map-deep-get($config-global, "spacing", "horizontal")};
	}
}

// Constrain left-aligned blocks
@media only screen and (min-width: 640px) {
	.entry-content > .wp-block-button:not(.alignleft):not(.alignright),
	.entry-content > .alignleft {
		margin-left: calc( 0.5 * (100% - calc( 640px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}

// Constrain right-aligned blocks
@media only screen and (min-width: 640px) {
	.entry-content > .alignright {
		margin-right: calc( 0.5 * (100% - calc( 640px - 32px)));
		max-width: calc( 0.5 * calc( 640px - 32px));
	}
}

// Constrain left-aligned blocks
@media only screen and (min-width: 782px) {
	.entry-content > .wp-block-button:not(.alignleft):not(.alignright),
	.entry-content > .alignleft {
		margin-left: calc( 0.5 * (100% - calc( 782px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}

// Constrain right-aligned blocks
@media only screen and (min-width: 782px) {
	.entry-content > .alignright {
		margin-right: calc( 0.5 * (100% - calc( 782px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}

// Constrain left-aligned blocks
@media only screen and (min-width: 1024px) {
	.entry-content > .wp-block-button:not(.alignleft):not(.alignright),
	.entry-content > .alignleft {
		margin-left: calc( 0.5 * (100% - calc( 782px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}

// Constrain right-aligned blocks
@media only screen and (min-width: 1024px) {
	.entry-content > .alignright {
		margin-right: calc( 0.5 * (100% - calc( 782px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}

// Constrain left-aligned blocks
@media only screen and (min-width: 1280px) {
	.entry-content > .wp-block-button:not(.alignleft):not(.alignright),
	.entry-content > .alignleft {
		margin-left: calc( 0.5 * (100% - calc( 782px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}

// Constrain right-aligned blocks
@media only screen and (min-width: 1280px) {
	.entry-content > .alignright {
		margin-right: calc( 0.5 * (100% - calc( 782px - 32px)));
		max-width: calc( 0.5 * calc( 782px - 32px));
	}
}
```

**Comment by itowhid06** [comment]
Hello @allancole,

I have updated the changes. Thank you very much for taking the time to reviewing and suggesting improvements.

**Pull request merged**